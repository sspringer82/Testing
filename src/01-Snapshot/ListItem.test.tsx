import ListItem from './ListItem';
import User from '../User';
import { fireEvent, render, screen } from '@testing-library/react';

const user: User = {
  id: 1,
  firstname: 'John',
  lastname: 'Doe',
};

describe('ListItem', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <ListItem user={user} onDelete={jest.fn()} />
        </tbody>
      </table>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
