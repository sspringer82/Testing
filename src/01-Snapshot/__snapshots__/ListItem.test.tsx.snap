// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match the snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          data-testid="id"
        >
          1
        </td>
        <td
          data-testid="firstname"
        >
          John
        </td>
        <td
          data-testid="lastname"
        >
          Doe
        </td>
        <td>
          <button
            data-testid="delete-btn"
          >
            l√∂schen
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
